sqlglot/dataframe/sql/column.py

```diff
@@ -0,0 +1,320 @@
+                exp.Column(this=exp.StrToTime(this=datetime_literal, format=exp.Literal.string("YYYY-MM-DD HH:MM:SS")))
+            )
+        if isinstance(value, datetime.date):
+            date_literal = exp.Literal.string(value.strftime("%Y-%m-%d"))
+            return cls(exp.Column(this=exp.StrToDate(this=date_literal, format=exp.Literal.string("YYYY-MM-DD"))))
+        return cls(value)
+
+    @classmethod
+    def invoke_anonymous_function(cls, column: t.Union[ColumnOrName, None], func_name: str, *args) -> Column:
+        column = [] if column is None else [cls.ensure_col(column)]
+        args = [cls.ensure_col(arg) for arg in args]
+        expressions = [x.expression for x in column + args]
+        new_expression = exp.Anonymous(this=func_name.upper(), expressions=expressions)
+        return Column(new_expression)
+
+    @classmethod
+    def invoke_expression_over_column(
+        cls, column: t.Union[ColumnOrName, None], callable_expression: t.Callable, **kwargs
+    ) -> Column:
+        column = None if column is None else cls.ensure_col(column)
+        new_expression = (
+            callable_expression(**kwargs)
+            if column is None
+            else callable_expression(this=column.column_expression, **kwargs)
+        )
+        return Column(new_expression)
+
+    def binary_op(self, klass: t.Callable, other: Column, **kwargs) -> Column:
+        return Column(klass(this=self.column_expression, expression=Column(other).column_expression, **kwargs))
+
+    def inverse_binary_op(self, klass: t.Callable, other: Column, **kwargs) -> Column:
+        return Column(klass(this=Column(other).column_expression, expression=self.column_expression, **kwargs))
+
+    def unary_op(self, klass: t.Callable, **kwargs) -> Column:
+        return Column(klass(this=self.column_expression, **kwargs))
+
+    @property
+    def is_alias(self):
+        return isinstance(self.expression, exp.Alias)
+
+    @property
+    def is_column(self):
+        return isinstance(self.expression, exp.Column)
+
+    @property
+    def column_expression(self) -> exp.Column:
+        if self.is_alias:
+            return self.expression.this
+        return self.expression
+
+    @property
+    def alias_or_name(self) -> str:
+        return self.expression.alias_or_name
+
+    @classmethod
+    def ensure_literal(cls, value) -> Column:
+        from sqlglot.dataframe.sql.functions import lit
+
+        if isinstance(value, cls):
+            value = value.expression
+        if not isinstance(value, exp.Literal):
+            return lit(value)
+        return Column(value)
+
+    def copy(self) -> Column:
+        return Column(self.expression.copy())
+
+    def set_table_name(self, table_name: str, copy=False) -> Column:
+        expression = self.expression.copy() if copy else self.expression
+        expression.set("table", exp.to_identifier(table_name))
+        return Column(expression)
+
+    def sql(self, **kwargs) -> Column:
```

Comment:
allow to pass in dialect too?

sql(**{"dialect": "spark", **kwargs})

Reply:
awesome great addition